src/colorizer/AnnotationData.ts

```diff
@@ -235,12 +257,55 @@ export class AnnotationData implements IAnnotationData {
     return idsToLabels;
   }
 
+  getValueFromId(labelIdx: number, id: number): string | null {
+    // This lookup may be too slow. May need to cache lookup from ID to value.
+    this.validateIndex(labelIdx);
+    const labelData = this.labelData[labelIdx];
+    if (labelData.ids.has(id)) {
+      // Search through values to find the one that matches the ID.
+      for (const [value, ids] of labelData.valueToIds) {
+        if (ids.has(id)) {
+          return value;
+        }
+      }
+      throw new Error(
+        "AnnotationData.getValueFromId: ID was listed as being labeled by label " +
+          `#${labelIdx} ('${labelData.options.name}'), but no value for ID ${id} was ` +
+          "found in the label data. This indicates a mismatch between the value lookup " +
+          " and the labeled IDs, and is likely a developer error."
+      );
+    } else {
+      return null;
+    }
+  }
+
   getNextDefaultLabelSettings(): LabelOptions {
     const color = new Color(
       DEFAULT_ANNOTATION_LABEL_COLORS[this.numLabelsCreated % DEFAULT_ANNOTATION_LABEL_COLORS.length]
     );
     const name = `Label ${this.numLabelsCreated + 1}`;
-    return { name, color };
+    return { type: LabelType.BOOLEAN, name, color, autoIncrement: true };
```

Comment:
Interesting how you have autoincrement true for this boolean but in the code right below here, you don't check the autoincrement flag in the boolean case!  (Booleans TRULY don't care about autoincrement!)